primitives/container-chain-genesis-data/src/lib.rs

```diff
@@ -68,18 +74,48 @@ pub mod json;
 )]
 #[serde(bound = "")]
 pub struct ContainerChainGenesisData {
-    pub storage: Vec<ContainerChainGenesisDataItem>,
-    // TODO: make all these Vec<u8> bounded
-    #[serde(with = "sp_core::bytes")]
-    pub name: Vec<u8>,
-    #[serde(with = "sp_core::bytes")]
-    pub id: Vec<u8>,
-    pub fork_id: Option<Vec<u8>>,
-    #[serde(with = "sp_core::bytes")]
-    pub extensions: Vec<u8>,
+    pub storage: BoundedVec<ContainerChainGenesisDataItem, ConstU32<655350>>,
+    #[serde(
+        serialize_with = "serialize_bounded_vec",
+        deserialize_with = "deserialize_bounded_vec"
+    )]
+    pub name: BoundedVec<u8, ConstU32<32>>,
+    #[serde(
+        serialize_with = "serialize_bounded_vec",
+        deserialize_with = "deserialize_bounded_vec"
+    )]
+    pub id: BoundedVec<u8, ConstU32<32>>,
+    pub fork_id: Option<BoundedVec<u8, ConstU32<32>>>,
+    #[serde(
+        serialize_with = "serialize_bounded_vec",
+        deserialize_with = "deserialize_bounded_vec"
+    )]
+    pub extensions: BoundedVec<u8, ConstU32<32>>,
     pub properties: Properties,
 }
 
+fn serialize_bounded_vec<S, const N: u32>(
```

Comment:
These function names should mention that it serializes as hex or bytes, maybe like
```suggestion
fn serialize_bounded_vec_as_hex<S, const N: u32>(
```

Reply:
Updated